examples/BERT/dist_model.py

```diff
@@ -38,77 +38,62 @@ def parameter_rrefs(self):
         return [RRef(p) for p in self.parameters()]
 
 
-class CrossLingualMLMTaskShard1(CrossLingualMLMTaskBase):
-    def __init__(self, device, ntoken, ninp, nhead, nhid, nlayers, dropout=0.5):
-        super(CrossLingualMLMTaskShard1, self).__init__(device)
-        self.xlmr_embed = XLMREmbedding(ntoken, ninp, dropout).to(device)
-        encoder_layers = TransformerEncoderLayer(ninp, nhead, nhid, dropout)
-        self.transformer_encoder = TransformerEncoder(encoder_layers, nlayers // 2).to(device)
-
-    def _forward(self, src):
-        output = self.xlmr_embed(src)
-        output = self.transformer_encoder(output)
-        return output
-
-
-class CrossLingualMLMTaskShard2(CrossLingualMLMTaskBase):
-    def __init__(self, device, ntoken, ninp, nhead, nhid, nlayers, dropout=0.5):
-        super(CrossLingualMLMTaskShard2, self).__init__(device)
-        encoder_layers = TransformerEncoderLayer(ninp, nhead, nhid, dropout)
-        self.transformer_encoder = TransformerEncoder(encoder_layers, nlayers // 2).to(device)
-        self.mlm_span = Linear(ninp, ninp).to(device)
-        self.activation = F.gelu
-        self.norm_layer = LayerNorm(ninp, eps=1e-12).to(device)
-        self.mlm_head = Linear(ninp, ntoken).to(device)
-
-    def _forward(self, src):
-        output = self.transformer_encoder(src)
-        output = self.mlm_span(output)
-        output = self.activation(output)
-        output = self.norm_layer(output)
-        output = self.mlm_head(output)
-        return output
-
-
 class DistCrossLingualMLMTask(nn.Module):
     """Two shards CrossLingualMLMTask"""
 
-    def __init__(self, split_size, workers, *args, **kwargs):
+    def __init__(self, nshards, split_size, workers, ntoken, ninp, nhead, nhid, nlayers, dropout=0.5, *args, **kwargs):
         super(DistCrossLingualMLMTask, self).__init__()
 
         self.split_size = split_size
 
-        # Put the first part of the ResNet50 on workers[0]
-        self.p1_rref = rpc.remote(
-            workers[0],
-            CrossLingualMLMTaskShard1,
-            args=(get_cuda_if_available(0),) + args,
-            kwargs=kwargs
-        )
-
-        # Put the second part of the ResNet50 on workers[1]
-        self.p2_rref = rpc.remote(
-            workers[1],
-            CrossLingualMLMTaskShard2,
-            args=(get_cuda_if_available(1),) + args,
-            kwargs=kwargs
-        )
+        encoder_layers = TransformerEncoderLayer(ninp, nhead, nhid, dropout)
+
+        self.shards = []
```

Comment:
Quick question, can we pass shards to the constructor? a.k.a. building the shards Line 53-77 outside the class constructor.